pkg/pfconfig/conversion.go

```diff
@@ -0,0 +1,150 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pfconfig
+
+import (
+	"fmt"
+	"math"
+	"net"
+	"strconv"
+	"time"
+
+	"go.thethings.network/lorawan-stack/pkg/band"
+	"go.thethings.network/lorawan-stack/pkg/frequencyplans"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+)
+
+func formatFrequency(frequency uint64) string {
+	freq := float64(frequency) / 1000000
+	if freq*10 == math.Floor(freq*10) {
+		return fmt.Sprintf("%.1f", freq)
+	}
+	return fmt.Sprintf("%g", freq)
+}
+
+// Build builds a packet forwarder configuration for the given gateway, using the given frequency plan store.
+func Build(gateway *ttnpb.Gateway, store *frequencyplans.Store) (*Config, error) {
```

Comment:
Adding to the discussion about making this package more generic, its preferable to split the building for `SX1301Conf` so that both BasicStation and PacketForwarder can call the `Build` method of `SX1301Conf` internally.
Please check the comment below on making this package more generic for BS and PF.

Reply:
The scope of this package (as the name indicates and the package doc says) is to "implement the JSON configuration of Semtech's reference Packet Forwarder". We have a separate basicstation package and we'll have a separate package for The Things Gateway, which also has a different config method. For those packages it would indeed be helpful to only use the SX1301 part of the config, so I splitted that functionality into a separate fun.

Reply:
Agreed. But can we create a "gatewayconfig" package and have sub-packages for each type?

Reply:
Right now that doesn't make sense, because we don't actually have any of those in `master` yet. But if you want, we can move this functionality when we do have multiple types.